File: donghwa/0x07/10866.py
import sys
from collections import deque

N = int(sys.stdin.readline().strip())
cmds = [sys.stdin.readline().strip() for _ in range(N)]
DQ = deque([])
for cmd in cmds:
    if cmd[:10] == "push_front":
        DQ.appendleft(int(cmd.split()[-1]))
    elif cmd[:9] == "push_back":
        DQ.append(int(cmd.split()[-1]))
    elif cmd == "pop_front":
        if (len(DQ)>0) :
            print(DQ.popleft())
        else : print(-1)
    elif cmd == "pop_back":
        if (len(DQ)>0) :
            print(DQ.pop())
        else : print(-1)
    elif cmd == "size":
        print(len(DQ))
    elif cmd == "empty":
        if (len(DQ) > 0) :
            print(0)
        else :
            print(1)
    elif cmd == "front":
        if (len(DQ) > 0) :
            print(DQ[0])
        else : print(-1)
    elif cmd == "back":
        if (len(DQ) > 0) :
            print(DQ[-1])
        else : print(-1)
    

File: donghwa/0x07/10866.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int N;
    cin >> N;
    deque<int> DQ;
    while(N--){
        string cmd;
        cin >> cmd;
        if (cmd == "push_front"){
            int x; cin >> x;
            DQ.push_front(x);
        } else if (cmd == "push_back"){
            int x; cin >> x;
            DQ.push_back(x);
        } else if (cmd == "pop_front"){
            if (DQ.size() > 0){
                cout << DQ.front() << '\n';
                DQ.pop_front();
            } else cout << -1 << '\n';
        } else if (cmd == "pop_back"){
            if (DQ.size() > 0){
                cout << DQ.back() << '\n';
                DQ.pop_back();
            } else cout << -1 << '\n';
        } else if (cmd == "size") cout << DQ.size() << '\n';
        else if (cmd == "empty") cout << DQ.empty() << '\n';
        else if (cmd == "front")
            if (DQ.size() > 0) cout << DQ.front() << '\n';
            else cout << -1 << '\n';
        else if (cmd == "back"){
            if (DQ.size() > 0){
                cout << DQ.back() << '\n';
            } else cout << -1 << '\n';
        }
        
    }
}